red.js

```diff
@@ -101,8 +101,15 @@ if (parsedArgs.settings) {
             var settingsStat = fs.statSync(defaultSettings);
             if (settingsStat.mtime.getTime() <= settingsStat.ctime.getTime()) {
                 // Default settings file has not been modified - safe to copy
-                fs.copySync(defaultSettings,userSettingsFile);
-                settingsFile = userSettingsFile;
+                try {
+                    fs.copySync(defaultSettings,userSettingsFile);
+                    settingsFile = userSettingsFile;
+                }
+                catch (err) {
+                    console.log("Can't copy settings file. Is file system Read Only ?");
```

Comment:
This could look at `err.code` to see if it's an access issue - `EACCES`.
Keeping in mind we don't (currently) translate the messages in this file, I'd suggest a rewording:

     Failed to copy settings file to '${userSettingsFile}'
     Error: ${err.toString()}

If it is EACCES, then having a nudge about settings.readOnly is helpful... except then they don't have a local settings file to set readOnly in.... so telling them to set readOnly in settings.js alone won't necessarily help. Hmm.

Reply:
Done, but yes this is where I think we could maybe just warn but handle but then set the readOnly flag for the user. If others are embedding they have to handle their version of the settings file at their own top level - and they may not try/want to copy it over anyway.